src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
+    private final int indexMethod = stackTraceElements.length - 2;
+    private StackTraceElement element = stackTraceElements[indexMethod];
+    private Object[] elementData = new Object[CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            elementData = grow(elementData.length);
```

Comment:
why parameters here? And let's assign a new array to elementData in the method itself